src/components/manage_dao/edit_form.tsx

```diff
@@ -1,302 +1,659 @@
+        : 'Unsupported Network';
+
+      return networkName;
+    }
+
+    return undefined;
+  }, [editedFormData]);
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const toastRef = useRef<ToastId>();
+
+  const showInfoToast = (text: string) => {
+    toastRef.current = toast({
+      position: 'top',
+      render: () => (
+        <InfoToast
+          link={text}
+          close={() => {
+            if (toastRef.current) {
+              toast.close(toastRef.current);
+            }
+          }}
+        />
+      ),
+    });
+  };
+
+  const updateEditError = (
+    key: keyof SettingsForm,
+    error: string | undefined
+  ) =>
+    setEditError((err) => ({ ...err, [key]: error ? { error } : undefined }));
+
+  const updateFormData = (update: Partial<SettingsForm>) => {
+    // eslint-disable-next-line guard-for-in, no-restricted-syntax
+    for (const key in update) {
+      updateEditError(key as any, undefined);
+    }
+
+    if (editedFormData) {
+      setEditedFormData((current) => ({ ...current!, ...update }));
+    }
+  };
+
+  const hasError = (key: keyof SettingsForm) => !!editError[key];
+
+  const handleImageChange = (
+    key: 'image' | 'coverImage',
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    if (event.target.files && event.target.files[0]) {
+      const img = event.target.files[0];
+      updateFormData({ [key]: URL.createObjectURL(img).toString() });
+    }
+  };
+
+  const handlePartnerImageChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    index: number
+  ) => {
+
+    if (event.target.files && event.target.files[0]) {
+      const img = event.target.files[0];
+      const maxImageSize = 2
+      if(img.size / 1024 / 1024 <= maxImageSize) {
+        let oldPartners = [...partners!];
+        oldPartners[index].image = URL.createObjectURL(img);
+        setPartners(oldPartners);
+        setChangedPartners(true);
+      } else {
+        toastRef.current = toast({
+          position: 'top',
+          render: () => ErrorToast({
+            content: `Image size exceeds ${maxImageSize} MB`,
+            close: () => {
+              if(toastRef.current) {
+                toast.close(toastRef.current)
+              }
+            },
+          }),
+        })
+      }
+  }};
+
+  const handleSubmit = async() => {
+    if (changedAbout) {
+      const newAboutString = await JSON.stringify(
+        convertToRaw(newAbout.getCurrentContent())
+      );
+  
+    let newAboutHash = (await uploadToIPFS(newAboutString)).hash
+    updateFormData({ about: newAboutHash });
+    }
+
+    if (changedPartners || changedPartnersImage) {
+      let oldPartners = [...partners!]
+      let partnerImageHash = '';
+      await Promise.all(oldPartners.map(async (partner, index) => {
+        partnerImageHash = (await uploadToIPFS(partner?.image!)).hash
+        oldPartners[index].image = partnerImageHash;
+        updateFormData({ partners: oldPartners });
+      }))
+    }
```

Comment:
hey @adiwajshing - looking to add partners and an encoded about to the smart contract request but struggling to send everything at the same time. It seems like the `uploadToIPFS` function takes a while and by the time the `handleSubmit` function initiates the update request, the IPFS hash is still not stored on the `editedFormData`.

I've been stuck on this for a bit, would love if you could give me a hand here - what do you recommend? Should I handle the IPFS upload on the `useWorkspaceUpdate` hook instead?